app/build.gradle

```diff
@@ -165,12 +161,11 @@ android {
   compileSdkVersion 27
 
   defaultConfig {
-    minSdkVersion 14
+    minSdkVersion 15
     targetSdkVersion 27
     testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
     // See https://github.com/linkedin/dexmaker/issues/65 for why we need the following line.
     testInstrumentationRunnerArguments.notClass = 'com.android.dex.DexIndexOverflowException'
-    multiDexEnabled true
```

Comment:
Are we sure this isn't needed anymore?

Reply:
Can I get a screenshot of the clipping bar?
Download Over wifi should affect new downloads but the state will not propagate to already running ones.
Pausing is not supported in the DownloadManager API but if they have configured it to download over wifi only then when they lose wifi connection it will automatically pause.
The parallel nature depends on the overall availability of the device, I started about 10 downloads and saw them all progress at once.
If your storage is switched to fat32 we filter out any files above 4GB.

I am pretty confident that with the removal of guava and apache we are not near the dex limit anymore. I will run a method count.

Reply:
62K is the current method count, so closer than I would like. Probably for next release we should look at integrating R8 to use as a shrinker/obfuscater.

Reply:
I tried changing the setting mid download and it wouldn't download on wifi still.
I don't think we should be filtering out imo, that is going to leave us with hundreds of one star reviews asking why people can't see their favourite ZIMs anymore, we need to make it very clear why they can't download them. Maybe display but put a translucent overlay with text over the top or something.

It doesn't quite clip but its about half a centimetre higher than it should be.

![image](https://user-images.githubusercontent.com/2286937/57968380-0b06cc80-796a-11e9-8007-0a152aad0614.png)

Reply:
I just added a commit to correct the top padding.
Re:wifi only see DownloadManagerRequester: 
 `       setAllowedNetworkTypes(
          if (sharedPreferenceUtil.prefWifiOnly) {
            Request.NETWORK_WIFI
          } else {
            Request.NETWORK_MOBILE or Request.NETWORK_WIFI
      } `
This is what controls the connection type of a request we send to the downloadManager. Once a download is in the downloadManager we have no control over it aside from cancelling it.
The current behaviour of filtering out zim files on fat32 detection was approved by @kelson42 , I could see an additional warning dialog on a switch to a Fat32 storage eg "Warning you will not be able to download files larger than 4GB on the selected storage. Do you wish to continue? Yes/No"

Reply:
@macgills I agree with Isacc that we should provide something better than just mask the "big ZIM files" if they can not be written. But we should focus on releasing 2.5 as is IMO, like you said. Please open a ticket about that enhancement and we should really do it then.

Reply:
https://github.com/kiwix/kiwix-android/issues/1173 see new issue here on 3.0 milestone